internal/pkg/template/repository/manifest/records.go

```diff
@@ -156,3 +166,30 @@ func (v *TemplateRecord) DefaultVersionOrErr() (VersionRecord, error) {
 	}
 	return version, nil
 }
+
+func (v *TemplateRecord) CheckComponents(components *model.ComponentsMap) bool {
+	for _, component := range v.Requirements.Components {
+		if _, found := components.Get(keboola.ComponentID(component)); !found {
+			return false
+		}
+	}
+	return true
+}
+
+func (v *TemplateRecord) CheckFeatures(d keboola.FeaturesMap) bool {
+	for _, feature := range v.Requirements.Features {
+		if !d.Has(feature) {
+			return false
+		}
+	}
+	return true
+}
+
+func (v *TemplateRecord) HasBackend(projectBackends []string) bool {
+	for _, backend := range v.Requirements.Backends {
```

Comment:
Please, add here

`if len(v.Requirements.Backends) == 0 { return true }`, it covers both cases: `nil` and empty slice.